homeassistant/components/starline/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""The StarLine component."""
+import voluptuous as vol
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .account import StarlineAccount
+from .const import (
+    DOMAIN,
+    PLATFORMS,
+    SERVICE_UPDATE_STATE,
+    SERVICE_SET_SCAN_INTERVAL,
+    CONF_SCAN_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured StarLine."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the StarLine device from a config entry."""
+    account = StarlineAccount(hass, config_entry)
+    await account.update()
+    if not account.api.available:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN] = account
```

Comment:
Your config flow does not error when a user tries to setup a 2nd account. Because of that, you will need to store account as `hass.data[DOMAIN][config_entry.entry_id] = account` to be able to support multiple entries. Alternative is to error out in the config flow.